### ISSUE TITLE
[CHECKPOINT]체크포인트ㅠㅠㅠㅠㅠㅠㅠㅠㅠㅠㅠㅠㅠ하ㅠㅠㅠㅠㅠㅠㅠㅠㅠ

### ISSUE THREAD

**Issue by ghost** [PARKMIHE/AICOCO#43]
import tensorflow as tf
tf.set_random_seed(777)

sess = tf.InteractiveSession()
new_saver = tf.train.import_meta_graph('/content/gdrive/My Drive/Colab Notebooks/n_flowers/my_test_model2.ckpt.meta')
new_saver.restore(sess, '/content/gdrive/My Drive/Colab Notebooks/n_flowers/my_test_model2.ckpt')

tf.get_default_graph()

x = sess.graph.get_tensor_by_name("x:0")
y = sess.graph.get_tensor_by_name("y_true:0")
hypothesis = sess.graph.get_tensor_by_name("operation:0")
k=sess.graph.get_tensor_by_name("keep_prob:0")
result = sess.run(hypothesis, feed_dict={x:c_test_x6[0:1],k:1.0})
print(sess.run(tf.argmax(c_test_y6,1)))
print(result)
print(sess.run(tf.argmax(result,1)))

**Comment by ghost** [comment]
출처
https://lueseypid.tistory.com/248

**Comment by ghost** [comment]
x = tf.placeholder(tf.float32, shape=[None, img_size_flat], name='x')
x_image = tf.reshape(x, [-1, img_size, img_size, num_channels])
y_true = tf.placeholder(tf.float32, shape=[None, num_classes], name='y_true')
y_true_cls = tf.argmax(y_true, axis=1)

내코드로 설명을 하자면 일단 x랑 y_true에 이름을 설정해줘야돼 그래야 체크포인트에서 쓸 수 있나봐! 나는 이름을 x랑 y_true로 설정을 해서 체크포인트 코드에서도 get_tensor_by_name 부분에 x랑 y_true로 이름을 넣어줬어 그리고 저 이름 뒤에 :0 꼭 해줘야돼 안해주면 형식 안맞는다고 에러 떠ㅠㅠㅠ

그리고 만약에 keep_prob 설정했으면 keep_prob에도 keep_prob=tf.placeholder(tf.float32,name='keep_prob') 이런식으로 이름을 설정해줘야돼 그러고 체크포인트 코드에서 keep_prob을 받을 변수를 설정해줘서 세션 돌릴 때 같이 넣어줘야돼 내가 저걸 안해줘서 자꾸 에러가 났었더라구ㅠㅠㅠㅠ 그리고 밑에 출력하는 세줄은 위에서부터 정답 출력, 이미지당 확률 출력, 그리고 prediction 출력이야 해보다가 이해안가면 물어봐!

**Comment by ghost** [comment]
import tensorflow as tf
import cv2
import numpy as np
tf.set_random_seed(777)

sess = tf.InteractiveSession()
new_saver = tf.train.import_meta_graph('/content/gdrive/My Drive/Colab Notebooks/n_flowers/my_test_model5.ckpt.meta')
new_saver.restore(sess, '/content/gdrive/My Drive/Colab Notebooks/n_flowers/my_test_model5.ckpt')

tf.get_default_graph()
img = cv2.imread('/content/gdrive/My Drive/Colab Notebooks/flowers/cherryblossom/007001.jpg',cv2.IMREAD_GRAYSCALE)
img=np.reshape(img,[-1,784])
x = sess.graph.get_tensor_by_name("x:0")
y = sess.graph.get_tensor_by_name("y_true:0")
hypothesis = sess.graph.get_tensor_by_name("operation:0")
k=sess.graph.get_tensor_by_name("keep_prob:0")
result = sess.run(hypothesis, feed_dict={x:img,k:1.0})

print(result[0])
print(sess.run(tf.argmax(result,1)))

**Comment by ghost** [comment]
if folders == 'forsythia': return [1,0,0,0,0,0,0]
    elif folders == 'korearose': return [0,1,0,0,0,0,0]
    elif folders == 'sunflower': return [0,0,1,0,0,0,0]
    elif folders == 'rose': return [0,0,0,1,0,0,0]
    elif folders == 'lily': return [0,0,0,0,1,0,0]
    elif folders == 'roseofsharon': return [0,0,0,0,0,1,0]
    elif folders == 'tulip': return [0,0,0,0,0,0,1]

**Comment by ghost** [comment]
[drive-download-20190613T041306Z-001.zip](https://github.com/PARKMIHE/AICOCO/files/3284179/drive-download-20190613T041306Z-001.zip)

**Comment by ghost** [comment]
[drive-download-20190613T071100Z-001.zip](https://github.com/PARKMIHE/AICOCO/files/3284605/drive-download-20190613T071100Z-001.zip)
